drivers/i3c/Kconfig.nxp

```diff
@@ -0,0 +1,17 @@
+# Copyright (c) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+DT_COMPAT_NXP_MCUX_I3C := nxp,mcux-i3c
+
+module = I3C_MCUX
+module-str = i3c-mcux
+source "subsys/logging/Kconfig.template.log_config"
+
+config I3C_MCUX
+	bool "MCUX I3C driver"
+	depends on HAS_MCUX
+	select PINCTRL
```

Comment:
why select pinctrl here? better depends?

Reply:
`CONFIG_I2C_MCUX` also selects `PINCTRL` and I simply copied it from there.